src/browser/setupWorker/setupWorker.ts

```diff
@@ -142,8 +143,11 @@ export class SetupWorkerApi
           })
         },
       },
-      useFallbackMode:
-        !('serviceWorker' in navigator) || location.protocol === 'file:',
+      supports: {
+        serviceWorkerApi:
```

Comment:
This is a good chance to move `useFallbackMode` to `supports.serviceWorkerApi` since now we check for two things on the side of browser's support.